code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,387 @@
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 1, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
+
+/obj/item/melee/secsword/Destroy()
+	if(cell?.loc == src)
+		QDEL_NULL(cell)
+	return ..()
+
+/obj/item/melee/secsword/examine(mob/user)
+	. = ..()
+	if(!cell)
+		. += "<span class='notice'>The powercell has been removed!</span>"
+		return
+	. += "<span class='notice'>It is [round(cell.percent())]% charged.</span>"
+	if(round(cell.percent() < 100))
+		. += "<span class='notice'>Can be recharged with a recharger.</span>"
+
+/obj/item/melee/secsword/examine_more(mob/user)
+	. = ..()
+	. += "A simple, practical blade developed by Shellguard munitions for ‘enhanced’ riot control."
+	. += ""
+	. += "The Securiblade is a simple blade of lightweight silver plasteel, honed to a nano-thin edge and with a simple \
+	power pack built into the hilt. A multi-purpose option for the NT Officer with a flair for impracticality, the \
+	Securiblade serves just as well as a regular sword, as it does a stun weapon or even an energized, heated blade."
+	. += ""
+	. += "While not the most popular option among the officers of Nanotrasen’s security forces, the Securiblade has still been valued for the multiple options it \
+	presents in combat. Deactivated, it’s a simple sword, but powered, it can either be utilized as a useful stun weapon, or as a dangerous, heated blade \
+	that can inflict grievous burn wounds on any suspects unfortunate enough to meet an officer using it. Rest assured, the Securiblade is a reliable tool in the hands of a skilled officer."
+
+/obj/item/melee/secsword/update_icon_state()
+	if(!cell)
+		icon_state = "[base_icon]3"
+		item_state = "[base_icon]0"
+	else
+		icon_state = "[base_icon][state]"
+		item_state = "[base_icon][state]"
+
+
+/obj/item/melee/secsword/proc/link_new_cell(unlink = FALSE)
+	if(unlink)
+		return
+	else
+		cell = new(src)
+
+/obj/item/melee/secsword/emp_act(severity)
+	cell.use(round(cell.charge / severity))
+	update_icon()
+
+/obj/item/melee/secsword/get_cell()
+	return cell
+
+/obj/item/melee/secsword/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/stock_parts/cell))
+		return
+	var/obj/item/stock_parts/cell/C = I
+	if(cell)
+		to_chat(user, "<span class='warning'>[src] already has a cell!</span>")
+		return
+	if(C.maxcharge < stam_hitcost)
+		to_chat(user, "<span class='warning'>[src] requires a higher capacity cell!</span>")
+		return
+	if(!user.unEquip(I))
+		return
+	I.forceMove(src)
+	cell = I
+	to_chat(user, "<span class='notice'>You install [I] into [src].</span>")
+	update_icon()
+
+/obj/item/melee/secsword/screwdriver_act(mob/living/user, obj/item/I)
+	if(!cell)
+		to_chat(user, "<span class='warning'>There's no cell installed!</span>")
+		return
+	if(!I.use_tool(src, user, volume = I.tool_volume))
+		return
+
+	user.put_in_hands(cell)
+	to_chat(user, "<span class='notice'>You remove [cell] from [src].</span>")
+	cell.update_icon()
+	cell = null
```

Comment:
you do a lot of `cell = null` + `update_icon()` calls throughout this file, why not make a proc called `clear_cell()` which nulls the reference, updates the icon, and can even handle signal tracking (for example if the cell is rigged to explode and gets QDEL'd inside the sword!)